### PR TITLE
Modified for epub  extension

### PR DESCRIPTION
**Pull request by protocolog**
## Description
Added epub directory in packages.
Added epub app call from docker oc10.web.config and ocis.idp.config

## Related Issue

- Fixes <issue_link>

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] ...

### CODE REVIEW

File: dev/docker/oc10.web.config.json [hunk 1]
```diff
@@ -11,6 +11,7 @@
     "files",
     "preview",
     "text-editor",
+      "epub-viewer",
```
Comment (dschmidt) [inline dev/docker/oc10.web.config.json:14]: Wrong indention

File: dev/docker/ocis.web.config.json [hunk 2]
```diff
@@ -28,6 +28,7 @@
     "files",
     "text-editor",
     "pdf-viewer",
+      "epub-viewer",
```
Comment (dschmidt) [inline dev/docker/ocis.web.config.json:31]: Indention

File: packages/web-app-epub-viewer/l10n/.tx/config [hunk 3]
```diff
@@ -0,0 +1,10 @@
+[main]
+host = https://www.transifex.com
+
+[owncloud-web.pdf-viewer]
```
Comment (dschmidt) [inline packages/web-app-epub-viewer/l10n/.tx/config:4]:
```suggestion
[owncloud-web.epub-viewer]
```

Comment (dschmidt) [reply packages/web-app-epub-viewer/l10n/.tx/config:4]: Also needs adding to the l18n Makefile (in web-runtime iirc)

File: packages/web-app-epub-viewer/l10n/translations.json [hunk 4]
```diff
@@ -0,0 +1 @@
+{"ar":{"msg":"رسالة"},"bg":{"Loading PDF document":"Зареждане на PDF документ","msg":"msg","PDF Viewer":"Преглед на PDF"},"cs":{},"de":{"Loading PDF document":"Das PDF-Dokument wird geladen","msg":"Nachricht","PDF Viewer":"PDF Vorschau"},"es":{"Loading PDF document":"Cargando documento PDF","msg":"msg","PDF Viewer":"Visor PDF"},"fr":{"Loading PDF document":"Chargement du document PDF","msg":"msg","PDF Viewer":"Visionneuse PDF"},"gl":{"msg":"msx"},"he":{"Loading PDF document":"מסמך PDF נטען","msg":"הודעה","PDF Viewer":"מציג PDF"},"it":{"Loading PDF document":"Caricamento documento PDF","PDF Viewer":"Visualizzatore PDF"},"pl":{"Loading PDF document":"Ładowanie dokumentu PDF","msg":"wiad","PDF Viewer":"Podgląd PDF"},"ru":{"msg":"сообщение"},"sk":{"msg":"msg"},"sq":{"Loading PDF document":"Po ngarkohet dokument PDF","msg":"Anëtarë hapësire","PDF Viewer":"Parës PDF-sh"},"tr":{"Loading PDF document":"PDF dokümanı yükleniyor","msg":"msj","PDF Viewer":"PDF Görüntüleyici"}}
```
Comment (dschmidt) [inline packages/web-app-epub-viewer/l10n/translations.json:1]: Incorrect translations, better ship an empty file than one like this

File: packages/web-app-epub-viewer/src/App.vue [hunk 5]
```diff
@@ -0,0 +1,80 @@
+<template>
+  <main>
+    <app-top-bar :resource="resource" @close="closeApp" />
+    <loading-screen v-if="loading" />
+    <error-screen v-else-if="loadingError" />
+    <div v-else class="oc-height-1-1">
+      <object class="epub-viewer oc-width-1-1" :data="url" type="application/epub" />
```
Comment (dschmidt) [inline packages/web-app-epub-viewer/src/App.vue:7]: I doubt this works, but I guess you're not finished yet...

Comment (protocolog) [reply packages/web-app-epub-viewer/src/App.vue:7]: Can you help me here, I have solved all other problems. I have started localhost:9200 with docker and local cloud working.

Comment (dschmidt) [reply packages/web-app-epub-viewer/src/App.vue:7]: Well, you need to figure out, how to render an epub file in a browser. I assume you need some js library to render an epub file

Comment (protocolog) [reply packages/web-app-epub-viewer/src/App.vue:7]: I have added the js file please check it and installed EPUB with pnpm.
epubjs

Comment (dschmidt) [reply packages/web-app-epub-viewer/src/App.vue:7]: I have no idea what that screenshot is supposed to show me

**Review by dschmidt — CHANGES_REQUESTED** [review]

File: packages/web-app-epub-viewer/src/App.vue (hunk 5 above)
Comment (dschmidt) [reply packages/web-app-epub-viewer/src/App.vue:7]: Just adding a dependency to the library is not enough, you need to `import` it in your `App.vue` and make use of it.

You likely need to call `renderTo` in your `mounted()` hook and on file change.

**Comment by dschmidt** [comment]
Are you still interested in finishing this? Afaict your uni projectjis over?

**Pull request closed without merging**